schedule/schedule-appengine/src/test/java/com/jasify/schedule/appengine/spi/ActivityEndpointTest.java

```diff
@@ -0,0 +1,517 @@
+        endpoint.addActivityType(null, null);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testAddActivityTypeNoActivityType() throws Exception {
+        testActivityServiceFactory.replay();
+        JasAddActivityTypeRequest jasAddActivityTypeRequest = new JasAddActivityTypeRequest();
+        endpoint.addActivityType(newCaller(1, true), jasAddActivityTypeRequest);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testAddActivityTypeNoOrganization() throws Exception {
+        testActivityServiceFactory.replay();
+        JasAddActivityTypeRequest jasAddActivityTypeRequest = new JasAddActivityTypeRequest();
+        jasAddActivityTypeRequest.setActivityType(new ActivityType());
+        endpoint.addActivityType(newCaller(1, true), jasAddActivityTypeRequest);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testAddActivityTypeThrowsNotFoundExceptionOnGetOrganization() throws Exception {
+        try {
+            testActivityServiceFactory.replay();
+            testOrganizationServiceFactory.setUp();
+            OrganizationService organizationService = OrganizationServiceFactory.getOrganizationService();
+            Key organizationKey = Datastore.allocateId(Organization.class);
+            organizationService.getOrganization(organizationKey);
+            expectLastCall().andThrow(new EntityNotFoundException());
+            testOrganizationServiceFactory.replay();
+
+            JasAddActivityTypeRequest jasAddActivityTypeRequest = new JasAddActivityTypeRequest();
+            jasAddActivityTypeRequest.setActivityType(new ActivityType());
+            jasAddActivityTypeRequest.setOrganizationId(organizationKey);
+
+            endpoint.addActivityType(newCaller(55, true), jasAddActivityTypeRequest);
+        } finally {
+            testOrganizationServiceFactory.tearDown(false);
+        }
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testAddActivityTypeThrowsBadRequestException() throws Exception {
+        try {
+            testOrganizationServiceFactory.setUp();
+            OrganizationService organizationService = OrganizationServiceFactory.getOrganizationService();
+            Key organizationKey = Datastore.allocateId(Organization.class);
+            Organization organization = new Organization();
+            expect(organizationService.getOrganization(organizationKey)).andReturn(organization);
+            testOrganizationServiceFactory.replay();
+
+            ActivityService activityService = ActivityServiceFactory.getActivityService();
+            JasAddActivityTypeRequest jasAddActivityTypeRequest = new JasAddActivityTypeRequest();
+            jasAddActivityTypeRequest.setActivityType(new ActivityType());
+            jasAddActivityTypeRequest.setOrganizationId(organizationKey);
+            activityService.addActivityType(organization, jasAddActivityTypeRequest.getActivityType());
```

Comment:
I would have made a contructor with the arguments you need.  As long as you also create a public default constructor `public JasAddActivityTypeRequest(){}`  it's fine.